src/meta/src/hummock/manager/compaction/compaction_group_schedule.rs

```diff
@@ -711,3 +546,340 @@ impl HummockManager {
+        &self,
+        group: &TableGroupInfo,
+        compaction_group_config: CompactionGroup,
+    ) {
+        let group_max_size = (compaction_group_config.max_estimated_group_size() as f64
+            * self.env.opts.compaction_group_size_threshold) as u64;
+        let is_huge_hybrid_group =
+            group.group_size > group_max_size && group.table_statistic.len() > 1; // avoid split single table group
+        if is_huge_hybrid_group {
+            assert!(group.table_statistic.keys().is_sorted());
+            let mut accumulated_size = 0;
+            let mut table_ids = Vec::default();
+            for (table_id, table_size) in &group.table_statistic {
+                accumulated_size += table_size;
+                table_ids.push(*table_id);
+                // split if the accumulated size is greater than half of the group size
+                // avoid split a small table to dedicated compaction group and trigger multiple merge
+                if accumulated_size * 2 > group_max_size {
+                    let ret = self
+                        .move_state_tables_to_dedicated_compaction_group(
+                            group.group_id,
+                            &table_ids,
+                            0,
+                        )
+                        .await;
+                    match ret {
+                        Ok(split_result) => {
+                            tracing::info!(
+                                "split_huge_compaction_group success {:?}",
+                                split_result
+                            );
+                            self.metrics
+                                .split_compaction_group_count
+                                .with_label_values(&[&group.group_id.to_string()])
+                                .inc();
+                            return;
+                        }
+                        Err(e) => {
+                            tracing::error!(
+                                error = %e.as_report(),
+                                "failed to split_huge_compaction_group table {:?} from group-{}",
+                                table_ids,
+                                group.group_id
+                            );
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    pub async fn try_merge_compaction_group(
+        &self,
+        table_write_throughput: &HashMap<u32, VecDeque<u64>>,
+        group: &TableGroupInfo,
+        group_config: CompactionGroup,
+        next_group: &TableGroupInfo,
+        _next_group_config: CompactionGroup,
+        checkpoint_secs: u64,
+        created_tables: &HashSet<u32>,
+    ) -> Result<()> {
+        // TODO: remove this check after refactor group id
+        if group.group_id == StaticCompactionGroupId::StateDefault as u64
+            && next_group.group_id == StaticCompactionGroupId::MaterializedView as u64
+        {
+            return Ok(());
```

Comment:
Should we return an error instead of ok here? With logics in timer_task.rs L621-630, I think returning error is more suitable because it will advance the left group as well.